app_integrations/apps/slack.py

```diff
@@ -0,0 +1,277 @@
+        self._next_page = 1
+
+    @classmethod
+    def _endpoint(cls):
+        """Class method to return the endpoint to be used for this slack instance
+
+        Returns:
+            str: Path of the desired endpoint to query
+
+        Raises:
+            NotImplementedError: If the subclasses do not properly implement this method
+        """
+        raise NotImplementedError('Subclasses should implement the _endpoint method')
+
+    @classmethod
+    def _type(cls):
+        raise NotImplementedError('Subclasses should implement the _type method')
+
+    @classmethod
+    def _sleep_seconds(cls):
+        raise NotImplementedError('Subclasses should implement the _sleep_seconds method')
+
+    @classmethod
+    def service(cls):
+        return 'slack'
+
+    @classmethod
+    def date_formatter(cls):
+        """Slack API date format: unix epoch seconds"""
+        return '%s'
+
+    @classmethod
+    def _required_auth_info(cls):
+        """Required credentials for access to the workspace"""
+        return {
+            'auth_token': {
+                'description': ('The security token generated by installing an app. '
+                                'This should be a string of characters beginning with xoxp-'),
+                'format': re.compile(r'^xoxp-[a-zA-Z0-9-]+$')
+                }
+            }
+
+    def _check_for_more_to_poll(self, response):
+        self._next_page += 1
+        return not ('paging' in response.keys() and
+                    response['paging']['pages'] == response['paging']['page'])
+
+    def _filter_response_entries(self, response):
+        """The slack endpoints don't provide a programatic way to filter for new results,
+        so subclasses must implement their own endpoint specific filtering methods"""
+        raise NotImplementedError("Subclasses must implement the _filter_response_entries method")
+
+    def _get_request_data(self):
+        '''The Slack API takes additional parameters to its endpoints in the body of the request.
+        Pagination control is one set of parameters.
+        '''
+        return {
+            'count': self._SLACK_API_MAX_ENTRY_COUNT,
+            'page': self._next_page
+        }
+
+    def _gather_logs(self):
+        """Gather log events.
+
+        Returns:
+            list: A list of dictionaries containing log events.
+        """
+        url = '{}{}'.format(self._SLACK_API_BASE_URL, self._endpoint())
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Authorization': 'Bearer {}'.format(self._config.auth['auth_token']),
+            }
+
+        data = self._get_request_data()
+
+        success, response = self._make_post_request(
+            url, headers, data, False)
+
+        if not success:
+            LOGGER.exception('Received bad response from slack')
+            return False
+
+        if not u'ok' in response.keys() or not response[u'ok']:
```

Comment:
can you omit the unicode `u` prefix here (and elsewhere). technically it's fine, just overly verbose.

also, to be more consistent with elsewhere, the can this please be changed to:

`if 'ok' not in response or not response['ok']`...

or: `if not response.get('ok')` would even do